homeassistant/components/event/__init__.py

```diff
@@ -0,0 +1,174 @@
+"""Component for handling incoming events as a platform."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+from typing import Any, final
+
+from homeassistant.backports.enum import StrEnum
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.config_validation import (  # noqa: F401
+    PLATFORM_SCHEMA,
+    PLATFORM_SCHEMA_BASE,
+)
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util import dt as dt_util
+
+from .const import ATTR_EVENT_TYPE, ATTR_EVENT_TYPES, DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EventDeviceClass(StrEnum):
+    """Device class for events."""
+
+    DOORBELL = "doorbell"
```

Comment:
Could we add `phone` here? I think the event platform could make sense for `fritzbox_callmonitor`. It currently uses a sensor but that isn't ideal. The event_types could be something like `ringing`, `dialing`, `talking`, and `idle` in that case.

Reply:
in `bthome`, we currently have `button` and `dimmer` events, of which `button` is used by the Shelly BLU button. 

![image](https://github.com/home-assistant/core/assets/24375847/f91aa80b-5a4f-4672-b801-e5cb15ae61fc)

xiaomi-ble also has events. 

![image](https://github.com/home-assistant/core/assets/24375847/657fc06c-567f-47b6-bb8b-1cdc58fe81fe)

So, I suggest to add the `dimmer` event. The xiaomi dimmers only report `rotate 3 clicks to the left`, etc. No state, just how much you rotate it from its current position. 
For the xiaomi `switch` events it is similar, xiaomi wall switches report `toggle` and the type of click, e.g. `left click, long click`. There is no way to tell it is `on` or `off` (no state). But we can also use `button` events for that.

Reply:
> Could we add `phone` here? I think the event platform could make sense for `fritzbox_callmonitor`. It currently uses a sensor but that isn't ideal. The event_types could be something like `ringing`, `dialing`, `talking`, and `idle` in that case.

@cdce8p We generally only add stuff like that if there is a wider/broader use case (multiple integrations are able to use it).

Not sure if that is the case here. That said, even without device classes, the entity can still be used (including translations). Device classes don't limit use cases.

Reply:
In general, I think I like the current set/use cases. We can always open up an arch discussion to create more device classes in the future. Those are easy to discuss and make a decision on.

Reply:
> should we at least add `OCCUPANCY` since we have `MOTION`?

I never really understood the difference...

So, Google tells me:

> Whilst a motion detector will only tell you when someone sits down at a desk, an occupancy sensor can tell you how long they stay there, even if they're not moving at all.

With that in mind, occupancy cannot be an event?

Reply:
Seems to me that `OCCUPANCY`, as in the length of the occupation, is more like something to be used as `binary_sensor` as you want to measure the time the state was `occupied`. To announce the last `occupance` I would use a sensor, as it is something in the past the time you known it.